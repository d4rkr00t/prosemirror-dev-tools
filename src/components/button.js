import styled from "styled-components";

const Button = styled.button`
  background: ${({ isActive, theme }) => isActive ? theme.main80 : theme.main40};
  border: none;
  color: #fff;
  padding: ${({ isSmall }) => isSmall ? "8px 16px;" : "12px 24px;"}
  text-transform: uppercase;
  transition: background .3s;
  border-radius: 3px;

  &:hover {
    background: ${({ isActive, theme }) => isActive ? theme.main90 : theme.main60};
    cursor: pointer;
  }

  &:focus {
    outline: none;
  }

  &:active {
    background: ${({ isActive, theme }) => isActive ? theme.main : theme.main80};
  }
`;

export default Button;
