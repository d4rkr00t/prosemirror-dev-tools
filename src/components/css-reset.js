import styled from "@emotion/styled";

const CSSReset = styled("div")({
  fontSize: "100%",
  lineHeight: 1,

  "& li + li": {
    margin: 0,
  },
});
CSSReset.displayName = "CSSReset";

export default CSSReset;
