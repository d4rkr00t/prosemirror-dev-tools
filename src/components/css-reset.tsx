import React from "react";
import { css } from "@compiled/react";

const cssResetStyles = css({
  fontSize: "100%",
  lineHeight: 1,

  "& li + li": {
    margin: 0,
  },
});

const CSSReset: React.FC = ({ children }) => {
  return <div css={cssResetStyles}>{children}</div>;
};

export default CSSReset;
