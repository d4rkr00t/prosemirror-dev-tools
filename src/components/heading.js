import styled from "styled-components";

const Heading = styled.h2`
  padding: 0;
  margin: 0;
  font-weight: 400;
  letter-spacing: 1px;
  font-size: 13px;
  color: ${props => props.theme.softerMain};
  text-transform: uppercase;
  flex-grow: 1;
`;

const HeadingWithButton = styled.div`
  display: flex;
`;

const HeadingButton = styled.button`
  padding: 6px 10px;
  margin: -6px -10px 0 8px;
  font-weight: 400;
  letter-spacing: 1px;
  font-size: 11px;
  color: ${props => props.theme.white80};
  text-transform: uppercase;
  transition: background .3s, color .3s;
  border-radius: 2px;
  border: none;
  background: transparent;

  &:hover {
    background: ${props => props.theme.main40};
    color: ${props => props.theme.white};
    cursor: pointer;
  }

  &:focus {
    outline: none;
  }

  &:active {
    background: ${props => props.theme.main60};
  }
`;

export { Heading, HeadingWithButton, HeadingButton };
