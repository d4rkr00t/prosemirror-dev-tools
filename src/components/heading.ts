import styled from "@emotion/styled";
import theme from "../theme";

const Heading = styled("h2")({
  color: theme.softerMain,
  padding: 0,
  margin: 0,
  fontWeight: 400,
  letterSpacing: "1px",
  fontSize: "13px",
  textTransform: "uppercase",
  flexGrow: 1,
});
Heading.displayName = "Heading";

const HeadingWithButton = styled("div")({
  display: "flex",
});
HeadingWithButton.displayName = "HeadingWithButton";

const HeadingButton = styled("button")({
  padding: "6px 10px",
  margin: "-6px -10px 0 8px",
  fontWeight: 400,
  letterSpacing: "1px",
  fontSize: "11px",
  color: theme.white80,
  textTransform: "uppercase",
  transition: "background 0.3s, color 0.3s",
  borderRadius: "2px",
  border: "none",
  background: "transparent",

  "&:hover": {
    background: theme.main40,
    color: theme.white,
    cursor: "pointer",
  },

  "&:focus": {
    outline: "none",
  },

  "&:active": {
    background: theme.main60,
  },
});
HeadingButton.displayName = "HeadingButton";

export { Heading, HeadingWithButton, HeadingButton };
