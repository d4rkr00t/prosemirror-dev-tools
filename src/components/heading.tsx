import React, { MouseEventHandler } from "react";
import { css } from "@compiled/react";
import theme from "../theme";

const headingStyles = css({
  color: theme.softerMain,
  padding: 0,
  margin: 0,
  fontWeight: 400,
  letterSpacing: "1px",
  fontSize: "13px",
  textTransform: "uppercase",
  flexGrow: 1,
});
const Heading: React.FC = ({ children }) => (
  <h2 css={headingStyles}>{children}</h2>
);

const headingWithButtonStyles = css({ display: "flex" });
const HeadingWithButton: React.FC = ({ children }) => (
  <div css={headingWithButtonStyles}>{children}</div>
);

const headingButtonStyles = css({
  padding: "6px 10px",
  margin: "-6px -10px 0 8px",
  fontWeight: 400,
  letterSpacing: "1px",
  fontSize: "11px",
  color: theme.white80,
  textTransform: "uppercase",
  transition: "background 0.3s, color 0.3s",
  borderRadius: "2px",
  border: "none",
  background: "transparent",

  "&:hover": {
    background: theme.main40,
    color: theme.text,
    cursor: "pointer",
  },

  "&:focus": {
    outline: "none",
  },

  "&:active": {
    background: theme.main60,
  },
});
const HeadingButton: React.FC<{
  onClick: MouseEventHandler<HTMLButtonElement>;
}> = ({ children, onClick }) => (
  <button onClick={onClick} css={headingButtonStyles}>
    {children}
  </button>
);

export { Heading, HeadingWithButton, HeadingButton };
