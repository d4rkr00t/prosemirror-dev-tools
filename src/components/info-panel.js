import styled from "@emotion/styled";
import theme from "../theme";

const InfoPanel = styled("div")({
  position: "relative",
  top: "50%",
  transform: "translateY(-50%)",
  textAlign: "center",
  color: theme.main,
  fontSize: "14px",
});
InfoPanel.displayName = "InfoPanel";

export { InfoPanel };
