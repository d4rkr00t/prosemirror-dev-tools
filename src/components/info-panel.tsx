import React from "react";
import { css } from "@compiled/react";
import theme from "../theme";

const infoPanelStyles = css({
  position: "relative",
  top: "50%",
  transform: "translateY(-50%)",
  textAlign: "center",
  color: theme.main,
  fontSize: "14px",
});
const InfoPanel: React.FC<React.PropsWithChildren> = ({ children }) => (
  <div css={infoPanelStyles}>{children}</div>
);

export { InfoPanel };
