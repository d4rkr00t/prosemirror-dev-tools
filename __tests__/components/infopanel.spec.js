import React from "react";
import renderer from "react-test-renderer";
import { InfoPanel } from "../../src/components/info-panel";

test("InfoPanel renders when active", () => {
  let info = renderer.create(<InfoPanel params="white" />).toJSON();
  expect(info).toMatchSnapshot();
});
