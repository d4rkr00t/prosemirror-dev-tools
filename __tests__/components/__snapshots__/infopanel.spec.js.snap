// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel renders when active 1`] = `
<div
  className="sc-bdVaJa fCfhyF"
/>
`;
